### DISCUSSION TITLE
Convergence error of a simple cantilever beam dynamics simulation

### DISCUSSION THREAD

**Discussion by xiaojbing** [idaholab/moose#27142] [Q&A Modules: Solid mechanics] [answered]
I tried to use the tensor-mechanics module in moose to solve a simple beam dynamics problem, but it seems to be some convergence problems.
**problem statement**：
This is a 3-D cantilever beam with a fixed left boundary end and a uniform downward pressure on the upper surface. I used the Newmark time integration method to simulate this problem, here is the geometric model of the beam and my input file:
![image](https://github.com/idaholab/moose/assets/87591264/17fe037f-786a-4a47-8d03-71a10ccf764c)
```
`
[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
[]
[Mesh]
  [generated_mesh]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 9
    ny = 3
    nz = 3

    xmin = 0
    xmax = 10
    ymin = 0
    ymax = 1
    zmin = 0
    zmax = 1
    boundary_name_prefix = bar
  []
[]

[Modules/TensorMechanics/DynamicMaster]
  [all]
    add_variables = true
    strain = FINITE
    density = 1000
    newmark_beta = 0.25
    newmark_gamma = 0.5
    hht_alpha = 0
  []
[]

[BCs]
  [left_x]
    type = DirichletBC
    variable = disp_x
    boundary = bar_left
    value = 0
  []
  [left_y]
    type = DirichletBC
    variable = disp_y
      boundary = bar_left
    value = 0
  []
  [left_z]
    type = DirichletBC
    variable = disp_z
      boundary = bar_left
    value = 0
  []
  [Pressure]
    [top _pre]
        boundary = bar_top
       function = 1e5      
       hht_alpha = 0
    []
  []
[]

[Materials]
  [elasticity]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 1e9
    poissons_ratio = 0.3
  []
  [stress]
    type = ComputeFiniteStrainElasticStress
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  line_search = none
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
  end_time = 10
  dt = 0.001
  nl_abs_tol = 1e-5
  [Predictor]
    type = SimplePredictor
    scale = 1
  []
  [./TimeIntegrator]
    type = NewmarkBeta
    inactive_tsteps = 1
  []
[]

[Outputs]
  csv = true

  checkpoint = false
  [console]
    type = Console
  []
  [exo]
    type = Exodus
    interval = 1
  []

[]

[Postprocessors]
  [point_A_disp_y]
    type = PointValue
    variable = disp_y
    point = '10 1 0'
  []
  [dt]
    type = TimestepSize
  []
  [nolinear_residual]
    type = Residual
  []
[]
`
```
**Error note**: 
The computation converges normally at the beginning, but as the computation proceeds, the nonlinear residuals end up getting larger and larger, and a decrease in the time step does not solve the problem.![image](https://github.com/idaholab/moose/assets/87591264/bdcd7b5f-ba6a-4c5c-85ae-d645d3239bf7)
![image](https://github.com/idaholab/moose/assets/87591264/ad4de201-5941-4531-b9c1-85a1acfd9d56)

**Comment by maxnezdyur** [comment, marked as answer]
Few ideas,
1) I think you are converging on the relative residual. Let's automatically scale and make the tolerance tighter.
```
  nl_abs_tol = 1e-10
  automatic_scaling = true 
```
2) Look at whether the problem needs damping. Newmark will grow unstable without any damping over time.

**Reply by xiaojbing** [reply]
thank you!! I will try.

**Reply by xiaojbing** [reply]
thank you! It works.
